src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3500,8 +3500,10 @@ private Message GetSortedSetAddMessage(RedisKey key, RedisValue member, double s
             }
         }
 
-        private Message GetSortedSetAddMessage(RedisKey destination, RedisKey key, long skip, long take, Order order, SortType sortType, RedisValue by, RedisValue[]? get, CommandFlags flags)
+        private Message GetSortMessage(RedisKey destination, RedisKey key, long skip, long take, Order order, SortType sortType, RedisValue by, RedisValue[]? get, CommandFlags flags)
         {
+            var command = destination.IsNull && multiplexer.GetServer(multiplexer.GetEndPoints()[0]).Version >= RedisFeatures.v7_0_0_rc1 ? RedisCommand.SORT_RO : RedisCommand.SORT;
```

Comment:
Here we'd want to add a new _internal_ member to `RedisFeatures`, e.g. `internal bool SortedSetPop => Version >= v7_0_0_rc1;` and use `var features = GetFeatures(key, flags, out ServerEndPoint? server);` like other places - note this means we need to pass server outward and use it upstream (to ensure we went to the server we said had the feature).

These types of switches are the _only_ things we want to be added to `RedisFeatures` overall but let's keep internal for now.

Reply:
Thanks, @NickCraver, that's much cleaner, I'll clean up my other PR that does that hocky junk as well.